src/components/Event/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { castDraft } from 'immer';
+import React, { useCallback, useContext, useState } from 'react';
+import {
+  faPencil,
+  faPalette,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons';
+
+import { classes, getContentClassName, periodToString } from '../../utils/misc';
+import { ActionRow, Palette } from '..';
+import { ScheduleContext } from '../../contexts';
+import { Period } from '../../types';
+
+import './stylesheet.scss';
+
+export type CourseProps = {
+  eventId: string;
+  eventName: string;
+  eventPeriod: Period;
+  eventDays: string[];
```

Comment:
I think we should make this readonly, since that is the type passed in by ScheduleContext.

Reply:
We actually don't need to. We can spread `eventDays` using the spread operator to remove the `readonly` tag. But no worries about it. I'll change that when I merge.

Reply:
Yes, we can. But I think we should maintain the type of eventDays.

Reply:
I understand what you're saying. But having a mix of `readonly` and non-`readonly` types in the props look kinda cumbersome I'd say. If we really want to maintain the type and keep the code clean, then it'd be best to just pass event id as props.

Reply:
that sounds like the best option